command/host_create.go

```diff
@@ -61,8 +66,94 @@ func (c *createHost) Execute(ctx context.Context, comm client.Communicator,
 			return err
 		}
 	}
+	startTime := time.Now()
+	ids, err := comm.CreateHost(ctx, taskData, *c.CreateHost)
+	if err != nil {
+		return errors.Wrap(err, "error creating host")
+	} else if c.CreateHost.CloudProvider != apimodels.ProviderDocker {
+		return nil // don't want to wait for logs
+	}
+
+	if len(ids) == 0 {
+		return errors.New("Programmer error: no intent host ID received")
+	}
+	if c.CreateHost.StderrFile == "" {
+		c.CreateHost.StderrFile = fmt.Sprintf("%s.err.log", ids[0])
+	}
+	if c.CreateHost.StdoutFile == "" {
+		c.CreateHost.StdoutFile = fmt.Sprintf("%s.out.log", ids[0])
+	}
+
+	if !c.CreateHost.Background {
+		return errors.Wrap(c.waitForLogs(ctx, comm, logger, startTime, ids[0]), "error waiting for logs")
+	}
+
+	go func() {
+		if err = c.waitForLogs(ctx, comm, logger, startTime, ids[0]); err != nil {
+			logger.Task().Errorf("error waiting for logs in background: %v", err)
+		}
+	}()
+
+	return nil
+}
+
+func (c *createHost) waitForLogs(ctx context.Context, comm client.Communicator, logger client.LoggerProducer,
+	startTime time.Time, hostID string) error {
+	pollTicker := time.NewTimer(time.Duration(c.CreateHost.PollFrequency) * time.Second)
+	defer pollTicker.Stop()
+
+	timeoutTimer := time.NewTimer(time.Duration(c.CreateHost.ContainerWaitTimeoutSecs) * time.Second)
+	defer timeoutTimer.Stop()
+
+	batchStart := startTime
+	// get logs in batches until container exits or we timeout
+	for {
+		select {
+		case <-ctx.Done():
+			return errors.New("context finished waiting for host to exit")
```

Comment:
Is a canceled context an error?

Reply:
Hmm I thought it would be. Could log something to task and return without an error instead? Or do we just want to return silently?